astropy/io/votable/tests/vo_test.py

```diff
@@ -820,6 +827,45 @@ def test_validate():
     assert truth == output
 
 
+@pytest.mark.skipif('not HAS_PATHLIB')
+def test_validate_path_object():
+    """
+    Validating when source is passed as path object. (#4412)
+    Creating different method from ``test_validate`` so that it could be
```

Comment:
I think you can just add an argument to `test_validate`, e.g.
```
def test_validate(test_path_object=False):
    fpath = get_pkg_data_filename('data/regression.xml')
    if test_path_object:
        fpath = pathlib.Path(fpath)
    ...
```
Then you can still have a separate `test_validate_path_object()` (which will be skipped for PY2) but just calls `test_validate(test_path_object=True)`.